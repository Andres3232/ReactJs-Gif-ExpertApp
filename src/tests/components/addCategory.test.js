import { shallow } from "enzyme"
import '@testing-library/jest-dom'
import React from 'react';
import { AddCategory } from "../../components/AddCategory";

describe('Prueebas en <AddCategory/>', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />); 

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />); 

    });

    test('debe mostrarse correctamentee', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'

        input.simulate('change',{target:{value}})
    })
    
    test('NO debe de psotear la info con submit', () => {
        
        wrapper.find('form').simulate('submit',{ preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled()
    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto ', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'

        //simular el inputChange
        input.simulate('change',{target:{value}})

        //simular el submit
        wrapper.find('form').simulate('submit',{ preventDefault(){} });

        //setCategories se debe de haber llamado
        expect( setCategories ).toHaveBeenCalled()
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) )

        //el valor del input debe de estar een ""
        expect( wrapper.find('input').prop('value') ).toBe('')

    }); 
    
    
});
