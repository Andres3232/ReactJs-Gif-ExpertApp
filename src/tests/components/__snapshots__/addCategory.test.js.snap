// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueebas en <AddCategory/> debe mostrarse correctamentee 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
